lib/msf/core/modules/external/python/metasploit/module.py

```diff
@@ -1,26 +1,37 @@
 import sys, os, json
 
+
 def log(message, level='info'):
     rpc_send({'jsonrpc': '2.0', 'method': 'message', 'params': {
         'level': level,
         'message': message
     }})
 
-def report_host(ip, opts={}):
+
+def report_host(ip, **opts):
     host = opts.copy()
     host.update({'host': ip})
     rpc_send({'jsonrpc': '2.0', 'method': 'report', 'params': {
         'type': 'host', 'data': host
     }})
 
-def report_service(ip, opts={}):
+
+def report_service(ip, **opts):
     service = opts.copy()
     service.update({'host': ip})
     rpc_send({'jsonrpc': '2.0', 'method': 'report', 'params': {
         'type': 'service', 'data': service
     }})
 
 
+def report_vuln(ip, name, **opts):
+    vuln = opts.copy()
+    vuln.update({'host': ip, 'name': name})
+    rpc_send({'jsonrpc': '2.0', 'method': 'report', 'params': {
```

Comment:
Minor: There is a bit of repetition between the various `report_*` methods in the first part of the JSON RPC message that you may consider pulling into a helper method at some point.